pcdsdevices/lodcm.py

```diff
@@ -84,13 +158,129 @@ class LODCM(BaseInterface, Device):
+    chi2_state = Cpt(CHI2, ':CHI2', kind='omitted')
+    # TODO energy component??
+    energy = Cpt(PseudoSingleInterface, egu='eV')
+
+    # Crystal Tower 1
+    z1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 Z')
+    x1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 X')
+    y1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 Y')
+    th1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 Theta')
+    ch1 = UCpt(IMS, kind='normal', doc='LOM Xtal1 Chi')
+    h1n_m = UCpt(IMS, kind='normal', doc='LOM Xtal1 Hn')
+    h1p = UCpt(IMS, kind='normal', doc='LOM Xtal1 Hp')
+    # Crystal Tower 2
+    z2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 Z')
+    x2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 X')
+    y2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 Y')
+    th2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 Theta')
+    ch2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 Chi')
+    h2n = UCpt(IMS, kind='normal', doc='LOM Xtal2 Hn')
+    diode2 = UCpt(IMS, kind='normal', doc='LOM Xtal2 PIPS')
+    # Diagnostic Tower
+    dh = UCpt(IMS, kind='normal', doc='LOM Dia H')
+    dv = UCpt(IMS, kind='normal', doc='LOM Dia V')
+    dr = UCpt(IMS, kind='normal', doc='LOM Dia Theta')
+    df = UCpt(IMS, kind='normal', doc='LOM Dia Filter Wheel')
+    dd = UCpt(IMS, kind='normal', doc='LOM Dia PIPS')
+    yag_zoom = UCpt(IMS, kind='normal', doc='LOM Zoom')
+    # reflection pvs
+    t1_c_ref = Cpt(EpicsSignalRO, ':T1C:REF', kind='omitted',
+                   doc='Tower 1 Diamond crystal reflection')
+    t1_si_ref = Cpt(EpicsSignalRO, ':T1Si:REF', kind='omitted',
+                    doc='Tower 1 Silicon crystal reflection')
+    t2_c_ref = Cpt(EpicsSignalRO, ':T2C:REF', kind='omitted',
+                   doc='Tower 2 Diamond crystal reflection')
+    t2_si_ref = Cpt(EpicsSignalRO, ':T2Si:REF', kind='omitted',
+                    doc='Tower 2 Silicon crystal reflection')
+    # pseudo positioners
+    th1_si = Cpt(PseudoSingleInterface, ':TH1:OFF_Si', kind='normal',
+                 name='th1_silicon', doc='Th1 motor offset for Si')
+    th1_c = Cpt(PseudoSingleInterface, ':TH1:OFF_C', kind='normal',
+                name='th1_diamond', doc='Th1 motor offset for C')
+    th2_si = Cpt(PseudoSingleInterface, ':TH2:OFF_Si', kind='normal',
+                 name='th2_silicon', doc='Th2 motor offset for Si')
+    th2_c = Cpt(PseudoSingleInterface, ':TH2:OFF_C', kind='normal',
+                name='th2_diamond', doc='Th2 motor offset for C')
+    z1_si = Cpt(PseudoSingleInterface, ':Z1:OFF_Si', kind='normal',
+                name='z1_silicon', doc='Z1 motor offset for Si')
+    z1_c = Cpt(PseudoSingleInterface, ':Z1:OFF_C', kind='normal',
+               name='z1_diamond', doc='Z1 motor offset for C')
+    z2_si = Cpt(PseudoSingleInterface, ':Z2:OFF_Si', kind='normal',
+                name='z1_diamond', doc='Z2 motor offset for Si')
+    z2_c = Cpt(PseudoSingleInterface, ':Z2:OFF_C', kind='normal',
+               name='z1_diamond', doc='Z2 motor offset for C')
```

Comment:
oh my this tower of pseudo positioners looks terrifying

Reply:
Yeah that's something I have a hard time wrapping my head around.
When we do the calculations, depending on what material we are using, we would be mapping different pseudo positioners. 
So something like this would not really work:

```
  if material == "Si":
            return self.PseudoPosition(th1_si=th, th2_si=th, z1_si=-z, z2_si=z)
  elif material == "C":
            return self.PseudoPosition(th1_c=th, th2_c=th, z1_c=-z, z2_c=z)
```
So we probably should have two different Energy subdevices in this case?...maybe...

Reply:
I think we should make it easier on ourselves by doing something like https://github.com/pcdshub/pcdsdevices/pull/715#discussion_r538970273